contrib/win32/win32compat/fileio.c

```diff
@@ -854,7 +854,7 @@ fileio_readlink_internal(wchar_t * wpath)
 	} 
 
 	/* append the symbolic link data to the output string*/
-	wcsncat(linkpath, symlink_nonnull, symlink_nonnull_size / sizeof(wchar_t));
+	wcsncat_s(linkpath, wcslen(linkpath), symlink_nonnull, symlink_nonnull_size / sizeof(wchar_t));
```

Comment:
If you're looking to use the _s version, I believe it needs to be the total size of the destination buffer in elements; wcslen(path) could be 0.  I'd recommend linkpath_len since that's what it's allocated to.